block/provider/list_snapshots.go

```diff
@@ -18,13 +18,76 @@
 package provider
 
 import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM/ibmcloud-volume-interface/lib/metrics"
 	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	userError "github.com/IBM/ibmcloud-volume-vpc/common/messages"
+	"github.com/IBM/ibmcloud-volume-vpc/common/vpcclient/models"
+	"go.uber.org/zap"
 )
 
+const startSnapshoIDNotFoundMsg = "start parameter is not valid"
+
 // ListSnapshots list all snapshots
-func (vpcs *VPCSession) ListSnapshots() ([]*provider.Snapshot, error) {
+func (vpcs *VPCSession) ListSnapshots(limit int, start string, filters map[string]string) (*provider.SnapshotList, error) {
 	vpcs.Logger.Info("Entry ListeSnapshots")
 	defer vpcs.Logger.Info("Exit ListSnapshots")
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "ListSnapshots", time.Now())
+
+	if limit < 0 {
+		return nil, userError.GetUserError("InvalidListSnapshotLimit", nil, limit)
+	}
+
+	if limit > maxLimit {
+		vpcs.Logger.Warn(fmt.Sprintf("listSnapshots requested max entries of %v, supports values <= %v so defaulting value back to %v", limit, maxLimit, maxLimit))
+		limit = maxLimit
+	}
+
+	filter := &models.LisSnapshotFilters{
```

Comment:
Please keep this is as TODO, I don't see this function getting called anywhere, the other ListSnapshots implamented for SnapshotService is getting called. But, in the future if this method is getting called anywhere. In line no. 50, 51. 52. map is being referenced without checking if those keys are present or not, which might crash. Please add a comment or todo anywhere. And just check where is the possibility for this function to be called.

Reply:
we are using this filters as of now also and related to In line no. 50, 51. 52. map is being referenced without checking if those keys are present or not, which might crash.
It won't crash as  if key in map is not present it gets zero value. So, making checks for all the key doesn't make sense.